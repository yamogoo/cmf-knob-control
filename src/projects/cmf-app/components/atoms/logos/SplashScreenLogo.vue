<template lang="pug">
lottie(
    ref="refRoot"
    :style="`width: ${100}%; height: ${100}%;`"
    :animation-data="Logo"
    :loop="false"
    @on-complete="onComplete"
)
</template>

<script setup lang="ts">
import Logo from '@app/assets/lottie/splashscreen-logo_02.json';

const emit = defineEmits<{
    (e: 'onComplete'): void
}>();

const onComplete = (): void => {
    emit('onComplete');
};
</script>

<style lang="scss"></style>