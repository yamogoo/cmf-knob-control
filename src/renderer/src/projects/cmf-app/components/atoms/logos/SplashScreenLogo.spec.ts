import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'

import SplashScreenLogo from './SplashScreenLogo.vue'

describe('SplashScreenLogo', () => {
  test('should render lottie icon DOM element', async () => {
    const component = mount(SplashScreenLogo)

    const lottieEl = component.find('[data-test-id="lottie"]')
    const isLottieElExist = lottieEl.exists()

    expect(isLottieElExist).toBe(true)
    expect(isLottieElExist).toMatchSnapshot()
  })
})
