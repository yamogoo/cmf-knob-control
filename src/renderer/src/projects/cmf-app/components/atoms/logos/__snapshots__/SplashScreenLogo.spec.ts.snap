// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SplashScreenLogo > should render lottie icon DOM element 1`] = `true`;
